cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/system/CDAPLogAppender.java

```diff
@@ -151,9 +152,11 @@ public void start() {
           Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("log-clean-up"));
         FileMetadataCleaner fileMetadataCleaner = new FileMetadataCleaner(context.getTransactionRunner());
         LogCleaner logCleaner = new LogCleaner(fileMetadataCleaner, context.getLocationFactory(),
+                                               logFileManager.getLogsDirectoryLocation(),
                                                TimeUnit.DAYS.toMillis(fileRetentionDurationDays),
                                                fileCleanupBatchSize);
         scheduledExecutorService.scheduleAtFixedRate(logCleaner, 10, logCleanupIntervalMins, TimeUnit.MINUTES);
+        scheduledExecutorService.execute(new LogFolderCleaner(logFileManager.getLogsDirectoryLocation(), 1000, 10000));
```

Comment:
Please use constants for this numbers. Even if not configurable, it helps to read the code. Otherwise one need to go to other class to see what those constants mean

Reply:
I am also a bit concerned you allow `start` to continue even before cleaning is finished. This potentially could produce collisions where main request servicing is running and tries to create a dir and this cleaner would remove it right away, potentially with data (if really unlucky and `isEmpty` said yes, but other thread created a file before deletion). It's better to do this on start synchronously before main processing started

Reply:
Constants for these literals is no longer required.

Executing the `Log Folder` cleaner first and then scheduling `Log File` cleaner. Since the executor is a single thread scheduled executor so no need for synchronous execution.